modules/internal/ChapelArray.chpl

```diff
@@ -3155,9 +3155,20 @@ module ChapelArray {
 
   pragma "init copy fn"
   proc chpl__initCopy(const ref rhs: [], definedConst: bool) {
-    pragma "no copy"
-    var lhs = chpl__coerceCopy(rhs.type, rhs, definedConst);
-    return lhs;
+    writeln("In initCopy(definedConst=", definedConst, ")", "domain definedConst: ", rhs.domain.definedConst);
+    if rhs.domain.definedConst && rhs.domain._value.locale != here {
```

Comment:
Just in case it helps save some time: I have bumped into different cases where `.locale` ended up to be a performance-killer. It has a very complex implementation including some dynamic dispatch if I am not remembering wrong. If you do time measurements or this is intended to optimize quick-running idioms, you might want to use `_wide_get_locale` primitive. I am doing that here: https://github.com/chapel-lang/chapel/pull/24390/files#diff-52d2477eb48a33d9e81b64d64d74a50145a1bacdd880825e267b5bd5549559afR2481-R2486. I think we should consider:

1. Improving `.locale`/`here` performance (there's discussion about this under some issues, but I can't find anything quickly. Note that `here` can also hurt performance) 
2. Having some helpers like `isLocal` at least internally for cases like that where we just use primitives for fast computation.